src/libraries/System.Runtime.Serialization.Formatters/tests/BinaryFormatterEventSourceTests.cs

```diff
@@ -0,0 +1,254 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Threading;
+using Xunit;
+
+namespace System.Runtime.Serialization.Formatters.Tests
+{
+    [ConditionalClass(typeof(PlatformDetection), nameof(PlatformDetection.IsBinaryFormatterSupported))]
+    public static class BinaryFormatterEventSourceTests
+    {
+        private const string BinaryFormatterEventSourceName = "System.Runtime.Serialization.Formatters.Binary.BinaryFormatterEventSource";
+
+        [Fact]
+        public static void RecordsSerialization()
+        {
+            LoggingEventListener listener = new LoggingEventListener();
```

Comment:
>LoggingEventListener listener = new LoggingEventListener(); [](start = 12, length = 59)

Is it better to convert this to 

using LoggingEventListener listener = new LoggingEventListener();

Just in case the test somehow throw before disposing?

Reply:
Throughout.

Reply:
The calls to `EventListener.Dispose` in each method are intended to unsubscribe the listener from all sources so that we don't receive any unexpected mutations to the log between the time we stop the test and the time we validate the results. I'll change the call site from `listener.Dispose()` to `var capturedLog = listener.Log.ToArray()` to make it explicit that we're intending to create a stable copy of the log.